lib/src/view/design_main_loopout_lengths.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:over_react/over_react.dart';
+import 'package:built_collection/built_collection.dart';
+import 'package:scadnano/src/state/geometry.dart';
+import 'package:scadnano/src/state/group.dart';
+import '../actions/actions.dart' as actions;
+import '../app.dart';
+import '../state/helix.dart';
+import '../util.dart';
+
+import '../state/strand.dart';
+import 'pure_component.dart';
+import '../constants.dart' as constants;
+import 'design_main_dna_sequence.dart';
+
+part 'design_main_loopout_lengths.over_react.g.dart';
+
+UiFactory<DesignMainLoopoutLengthsProps> DesignMainLoopoutLengths = _$DesignMainLoopoutLengths;
+
+mixin DesignMainLoopoutLengthsProps on UiProps {
+  BuiltMap<int, Helix> helices;
+  BuiltMap<String, HelixGroup> groups;
+  Geometry geometry;
+
+  BuiltList<Strand> strands;
+  BuiltSet<int> side_selected_helix_idxs;
+  String dna_sequence_png_uri;
+  bool is_zoom_above_threshold;
+  actions.Action disable_png_cache_until_action_completes;
+  bool only_display_selected_helices;
+  bool show_dna;
+  bool show_loopout_length;
+}
+
+class DesignMainLoopoutLengthsComponent extends UiComponent2<DesignMainLoopoutLengthsProps>
+    with PureComponent {
+  /// This method is used to handle the `disable_png_cache_until_action_completes` prop.
+  /// If this prop is not null, then it dispatches the prop action before disabling it.
+  @override
+  void componentDidUpdate(Map prevProps, Map prevState, [snapshot]) {
+    var action_to_complete = props.disable_png_cache_until_action_completes;
+
+    if (action_to_complete != null) {
+      app.dispatch(action_to_complete);
+
+      // After exporting svg, re-enable png caching.
+      app.dispatch(actions.SetDisablePngCacheUntilActionCompletes(null));
+    }
+  }
+
+  @override
+  render() {
+    if (props.show_loopout_length) {
+      return (Dom.g()..className = 'dna-sequences-main-view')([
+        for (Strand strand in props.strands)
+          if (strand.dna_sequence != null)
+            (DesignMainDNASequence()
```

Comment:
This isn't quite what I meant. I would completely separate the components that draw loopout lengths from those that draw DNA sequences. So here, it should be something like

```dart
for (Strand strand in props.strands)
  for (Loopout loopout in strand.loopouts())
    (DesignMainLoopout()
      // ...
```

where `DesignMainLoopout` is its own component that is only responsible for drawing loopout lengths, and there is no code in `DesignMainDNASequence` that has anything to do with drawing loopout lengths.